src/components/CreatureDetail/CreatureDetail.jsx

```diff
@@ -0,0 +1,87 @@
+import { useParams } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import './CreatureDetail.css';
+import { getCreatureByIndex } from '../../apiCalls';
+
+const CreatureDetail = () => {
+  const { index } = useParams();
+  const [selectedCreature, setSelectedCreature] = useState(null);
+  const [error, setError] = useState(null);
+
+  const renderElementIfExists = (label, value) => {
+    if (value) {
+      return (
+        <p>
+          <strong>{label}: </strong>
+          {value}
+        </p>
+      );
+    };
+    return null;
+  };
+
+  const renderArrayIfExists = (label, array) => {
+    if (array && array.length > 0) {
+      return (
+        <div>
+          <p>
+            <strong>{label} </strong>
+          </p>
+          {array.map((item) => (
+            <p key={item.name}>{item.name}: {item.desc}</p>
+          ))}
+        </div>
+      );
+    };
+    return null;
+  };
```

Comment:
@amysiu1028 Thank you for your super helpful note about my p tag in a p tag issue! I've pushed an update here. Does that look right now?

Reply:
@ericahagle awesome work! I went through the application just now and don't see any errors on the console!